### PR TITLE
Enable CUDA Python Test

### PR DESCRIPTION
**Pull request by jchen351**

### CODE REVIEW

File: onnxruntime/test/testdata/onnx_backend_test_series_filters.jsonc [hunk 1]
```diff
@@ -306,6 +306,11 @@
         "^test_qlinearmatmul_3D_int8_float32_cuda",
         "^test_qlinearmatmul_3D_uint8_float16_cuda",
         "^test_qlinearmatmul_3D_uint8_float32_cuda",
+        // Test that failed on CUDA 12.2.
+         "^test_Conv3d_dilated_cuda",
+        "^test_Conv3d_dilated_strided_cuda",
```
Comment (snnn) [inline onnxruntime/test/testdata/onnx_backend_test_series_filters.jsonc:311]: please format the file.  Line 310 has an extra space.

File: tools/ci_build/github/azure-pipelines/py-cuda-packaging-pipeline.yml [hunk 2]
```diff
@@ -24,6 +24,14 @@ parameters:
       - RelWithDebInfo
       - MinSizeRel
 
+  - name: cuda_version
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/py-cuda-packaging-pipeline.yml:27]: No need to make it configurable. We have two such pipelines.  We want to run them daily. So the version number needs be fixed.

File: tools/ci_build/github/azure-pipelines/stages/py-gpu-packaging-stage.yml [hunk 3]
```diff
@@ -79,6 +79,6 @@ stages:
           MACHINE_POOL: 'onnxruntime-Win2022-GPU-dml-A10'
           PYTHON_VERSION: ${{ python_version }}
           EP_BUILD_FLAGS: --use_dml --cmake_extra_defines CMAKE_SYSTEM_VERSION=10.0.18362.0 --enable_wcos
-          ENV_SETUP_SCRIPT: setup_env.bat
           EP_NAME: directml
-          cmake_build_type: ${{ parameters.cmake_build_type }}
\ No newline at end of file
+          cmake_build_type: ${{ parameters.cmake_build_type }}
+          CudaVersion: ${{ parameters.cuda_version }}
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/stages/py-gpu-packaging-stage.yml:84]: We should not need to set CUDA version when the EP name is DML?

Comment (jchen351) [reply tools/ci_build/github/azure-pipelines/stages/py-gpu-packaging-stage.yml:84]: That is correct. We can just use default.

File: tools/ci_build/github/azure-pipelines/stages/py-win-gpu-stage.yml [hunk 4]
```diff
@@ -211,6 +198,21 @@ stages:
...
+            CudaVersion: ${{ parameters.CudaVersion }}
+            ${{ if or(contains(parameters.EP_BUILD_FLAGS, 'use_cuda'), eq(parameters.use_tensorrt, true)) }}:
+              DownloadCUDA: true
+            DownloadTRT: ${{ parameters.use_tensorrt }}
+
+#        TODO: remove this task
+        - task: BatchScript@1
+          displayName: 'setup env'
+          inputs:
+            filename: '$(Build.SourcesDirectory)\tools\ci_build\github\windows\setup_env.bat'
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/stages/py-win-gpu-stage.yml:212]: This file is for GPU jobs.  And the "setup_env.bat" does not append any dir to PATH env.  I think we do not need  this step at all. (Or maybe a wrong bat file is used here)

**Review by snnn — APPROVED** [review]

**Pull request merged**